index.d.ts

```diff
@@ -20,6 +20,7 @@ console.log(count);
 ```
 */
 export default function pDoWhilst<ValueType>(
-	action: () => ValueType | PromiseLike<ValueType>,
-	condition: (value: ValueType) => boolean
-): Promise<void>;
+	action: (value: ValueType | PromiseLike<ValueType>) => ValueType | PromiseLike<ValueType>,
+	condition: (value: ValueType | PromiseLike<ValueType>) => boolean | PromiseLike<boolean>,
+	initialValue?: ValueType | PromiseLike<ValueType>,
```

Comment:
This is incorrect. It should preserve the input `ValueType`.

```suggestion
	action: (value: ValueType) => ValueType | PromiseLike<ValueType>,
	condition: (value: ValueType) => boolean | PromiseLike<boolean>,
	initialValue?: ValueType,
```

Reply:
Right! Fixed in cebefc8.